core.go

```diff
@@ -142,6 +143,20 @@ type (
 		qos       QOSLevel               // not exported to JSON
 	}
 
+	// BulkData represents an event to be sent to the MODE cloud.
+	DeviceBulkData struct {
+		Version    string
+		SeriesIDs  []string
+		DataBlocks []BulkDataBlock
+		qos        QOSLevel // not exported to serializer
+	}
+
+	// A single row of data.  Size of Values must match with size of SeriesIDs
+	BulkDataBlock struct {
+		TimeStamp time.Time
+		Values    []float64 // Size matches BulkData.SeriesIDs
+	}
+
```

Comment:
This structure is specific to time-series data. Instead, we want to keep the data as opaque blob. Just like "event" is basically eventType and opaque eventData, the bulk data will be "streamID" and an opaque data blob ([]byte).

The encoding of time-series data (into events and bulk data) should be outside of the core device-sdk package (but perhaps as a sub-package) since not all clients are using TSDB.

Reply:
done. please review.